gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_model.dart

```diff
@@ -2,14 +2,31 @@ import 'package:agentapi/agentapi.dart';
 import 'package:dart_either/dart_either.dart';
 import 'package:flutter/foundation.dart';
 import 'package:p4w_ms_store/p4w_ms_store.dart';
+
 import '/core/agent_api_client.dart';
 import '/core/pro_token.dart';
 
-class SubscribeNowModel {
+class SubscribeNowModel extends ChangeNotifier {
   final AgentApiClient client;
+  P4wMsStore store;
+
+  final _token = ProTokenValue();
+  ProTokenValue get token => _token;
+  bool get canSubmit => token.token != null;
```

Comment:
We can get rid of the ProTokenValue class and use the Either directly here. Or even completely remove the usage of eithers and do like we do in the Landscape page, where we store both the value and the error simultaneously. Personally, I prefer the Either monad, because if we have an error we don't want to keep the value.

```suggestion
  final _token = Either<TokenError,ProToken?>;
  ProToken? get token => _token.orNull();
  bool get canSubmit => token.isRight();
```